### ISSUE TITLE
How to use pi4j to write code in pcf8574

### ISSUE THREAD

**Issue by YohannHe** [Pi4J/pi4j-v1#221]
This my code：
`import java.io.IOException;
import com.pi4j.component.lcd.impl.GpioLcdDisplay;
import com.pi4j.gpio.extension.pcf.PCF8574GpioProvider;
import com.pi4j.gpio.extension.pcf.PCF8574Pin;
import com.pi4j.io.gpio.GpioController;
import com.pi4j.io.gpio.GpioFactory;
import com.pi4j.io.gpio.GpioPinDigitalOutput;
import com.pi4j.io.gpio.PinMode;
import com.pi4j.io.gpio.PinState;
import com.pi4j.io.gpio.RaspiPin;
import com.pi4j.io.i2c.I2CBus;

public class Test {
    public static void main(String[] args) throws IOException,InterruptedException  {

```
    final GpioController gpio = GpioFactory.getInstance();
    PCF8574GpioProvider gpioProvider;

        gpioProvider = new PCF8574GpioProvider(I2CBus.BUS_0,PCF8574GpioProvider.PCF8574_0x27);
        GpioPinDigitalOutput rsPin = gpio.provisionDigitalOutputPin(gpioProvider, PCF8574Pin.GPIO_04,PinState.LOW);
        GpioPinDigitalOutput o2 = gpio.provisionDigitalOutputPin(gpioProvider, PCF8574Pin.GPIO_05,PinState.LOW);
        GpioPinDigitalOutput strobe = gpio.provisionDigitalOutputPin(gpioProvider, PCF8574Pin.GPIO_06,PinState.LOW);
        GpioPinDigitalOutput d4 = gpio.provisionDigitalOutputPin(gpioProvider, PCF8574Pin.GPIO_00,PinState.LOW);
        GpioPinDigitalOutput d5 = gpio.provisionDigitalOutputPin(gpioProvider, PCF8574Pin.GPIO_01,PinState.LOW);
        GpioPinDigitalOutput d6 = gpio.provisionDigitalOutputPin(gpioProvider, PCF8574Pin.GPIO_02,PinState.LOW);
        GpioPinDigitalOutput d7 = gpio.provisionDigitalOutputPin(gpioProvider, PCF8574Pin.GPIO_03,PinState.LOW);
        GpioLcdDisplay lcd = new GpioLcdDisplay(2, 16, rsPin.getPin(), strobe.getPin(), d4.getPin(),d5.getPin(),d6.getPin(),d7.getPin());
        lcd.clear();
        lcd.setCursorHome();
        lcd.writeln(1,"Hello!");
        System.out.println("Done");




}
```

}
`

**Comment by savageautomate** [comment]
Well, it looks like the `GpioLcdDisplay` is not compatible with GPIO extensions such as the `PCF8574GpioProvider` implementation.  The underlying impl code in `GpioLcdDisplay` is making use of the WiringPi LCD functions which only work with the native (onboard) Raspberry Pi GPIO pins.  You would have to re-implement the bit-banging logic of the WiringPi LCD functions in Java and write the data to the `PCF8574GpioProvider` GPIO pins manually.  

Basically all the logic in this C code would need to be ported to Java.  https://github.com/Pi4J/wiringPi/blob/master/devLib/lcd.c

If you end up doing this and wish to contribute your work to the Pi4J project, we can include it.

Thanks, Robert
